components/admin_console/permission_schemes_settings/edit_post_time_limit_modal/edit_post_time_limit_modal.tsx

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import {Modal} from 'react-bootstrap';
+
+import {AdminConfig} from 'mattermost-redux/types/config';
+import {ActionFunc} from 'mattermost-redux/types/actions';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import {Constants} from 'utils/constants';
+import {localizeMessage} from 'utils/utils.jsx';
+import {t} from 'utils/i18n';
+
+const INT32_MAX = 2147483647;
+
+type Props ={
+    config: Partial<AdminConfig>;
+    show: boolean;
+    onClose: () => void;
+    actions: {
+        updateConfig: (config: AdminConfig) => ActionFunc & Partial<{error?: ClientErrorPlaceholder}>;
+        getConfig: () => ActionFunc;
+    };
+}
+
+type ClientErrorPlaceholder = {
+    message: string;
+    server_error_id: string;
+}
+
+export default function EditPostTimeLimitModal(props: Props) {
+    const [saving, setSaving] = React.useState(false);
+    const [errorMessage, setErrorMessage] = React.useState('');
+    const [postEditTimeLimit, setPostEditTimeLimit] = React.useState<number | string>(parseInt(props.config.ServiceSettings!.PostEditTimeLimit.toString(), 10));
+
+    React.useEffect(() => {
+        props.actions.getConfig();
+    }, []);
+
+    const save = async () => {
+        setSaving(true);
+        setErrorMessage('');
+
+        const val = parseInt(postEditTimeLimit.toString(), 10);
+        if (val !== Constants.UNSET_POST_EDIT_TIME_LIMIT) {
+            if (val.toString() === 'NaN' || val <= 0 || val > INT32_MAX) {
+                setErrorMessage(localizeMessage('edit_post.time_limit_modal.invalid_time_limit', 'Invalid time limit'));
+                setSaving(false);
+                return false;
+            }
+        }
+
+        const newConfig = JSON.parse(JSON.stringify(props.config));
+        newConfig.ServiceSettings.PostEditTimeLimit = val;
+
+        const {error} = await props.actions.updateConfig(newConfig);
+        if (error) {
+            setErrorMessage(error.message);
+            setSaving(false);
+        } else {
+            setSaving(false);
+            props.onClose();
+        }
+
+        return true;
+    };
+
+    const handleOptionChange = (e: React.FormEvent<HTMLInputElement>) => {
+        const {value} = e.currentTarget;
+        if (value === Constants.ALLOW_EDIT_POST_ALWAYS) {
+            setPostEditTimeLimit(Constants.UNSET_POST_EDIT_TIME_LIMIT);
+        } else {
+            setPostEditTimeLimit('');
+        }
+    };
+
+    const handleSecondsChange = (e: React.FormEvent<HTMLInputElement>) => {
+        const {value} = e.currentTarget;
+        setPostEditTimeLimit(value);
+    };
```

Comment:
This is the only place where we need to parse the value:

```suggestion
    const handleSecondsChange = ({currentTarget}: React.FormEvent<HTMLInputElement>) => setPostEditTimeLimit(parseInt(currentTarget.value, 10));
```

Reply:
Yep. Done